src/cli/eofwrap.py

```diff
@@ -0,0 +1,370 @@
+                    requests_root=header.requests_root,
+                )
+                assert not fixture_block.ommers
+                assert not fixture_block.withdrawals
+                assert not fixture_block.deposit_requests
+                assert not fixture_block.withdrawal_requests
+                assert not fixture_block.consolidation_requests
+
+                for fixture_tx in fixture_block.txs:
+                    tx = Transaction(
+                        type=fixture_tx.ty,
+                        chain_id=fixture_tx.chain_id,
+                        nonce=fixture_tx.nonce,
+                        gas_price=fixture_tx.gas_price,
+                        max_priority_fee_per_gas=fixture_tx.max_priority_fee_per_gas,
+                        max_fee_per_gas=fixture_tx.max_fee_per_gas,
+                        gas_limit=fixture_tx.gas_limit,
+                        to=fixture_tx.to,
+                        value=fixture_tx.value,
+                        input=fixture_tx.data,
+                        access_list=fixture_tx.access_list,
+                        max_fee_per_blob_gas=fixture_tx.max_fee_per_blob_gas,
+                        blob_versioned_hashes=fixture_tx.blob_versioned_hashes,
+                        v=fixture_tx.v,
+                        r=fixture_tx.r,
+                        s=fixture_tx.s,
+                        sender=fixture_tx.sender,
+                        authorization_list=fixture_tx.authorization_list,
+                    )
+                    block.txs.append(tx)
+            elif isinstance(fixture_block, InvalidFixtureBlock):
+                block = Block(
+                    rlp=fixture_block.rlp,
+                    exception=fixture_block.expect_exception,
+                )
+            else:
+                raise TypeError("not a FixtureBlock")
+
+            test.blocks.append(block)
+
+        return test.generate(
+            request=None,  # type: ignore
+            t8n=t8n,
+            fork=CancunEIP7692,
+            fixture_format=BlockchainFixture,
+        )
+
+    def _validate_eof(self, container: Container, metrics: bool = True) -> bool:
+        eof_parse = EOFParse()
+
+        result = eof_parse.run(input=to_hex(container))
+        actual_message = result.stdout.strip()
+        if "OK" not in actual_message:
+            if metrics:
+                _inc_counter(self.metrics[self.VALIDATION_ERRORS], actual_message)
+            return False
+
+        return True
+
+
+def _has_opcode_at(code: Bytecode, i: int, op: Op) -> bool:
+    opcode_at = OPCODE_MAP.get(
+        bytes(code)[i],
+    )
+    if not opcode_at:
+        return False
+
+    return opcode_at == op
+
+
+def wrap_code(account_code: Bytes) -> Container:
+    """
+    Wraps `account_code` into a simplest EOF container, applying some simple heuristics in
+    order to obtain a valid code section termination.
+    """
+    assert len(account_code) > 0
```

Comment:
I have the feeling that doing something like we do here would yield better results maybe:
https://github.com/ethereum/execution-spec-tests/blob/0900aa8ef592f968c119f11cdcd934eb2d360693/src/cli/evm_bytes.py#L68-L92

The stack would be automatically calculated for you and also you could check the `terminating` attribute of `Bytecode` to decide whether to add a `STOP` or not.

Reply:
`fix(cli): rework eofwrap using process_evm_bytes` accomplishes that, thanks for the suggestions. It required a bit  of fiddling and adding a parameter, but finally all JSON files went through.